src/home_robot_hw/home_robot_hw/env/stretch_pick_and_place_env.py

```diff
@@ -114,59 +123,76 @@ def apply_action(self, action: Action, info: Optional[Dict[str, Any]] = None):
             print("Action not implemented in pick-and-place environment:", action)
             continuous_action = None
 
         # Move, if we are not doing anything with the arm
-        if continuous_action is not None:
-            if not self.in_navigation_mode():
-                self.switch_to_navigation_mode()
+        if continuous_action is not None and not self.test_grasping:
+            print("Execute navigation action:", continuous_action)
+            if not self.robot.in_navigation_mode():
+                self.robot.switch_to_navigation_mode()
                 rospy.sleep(self.msg_delay_t)
-            self.navigate_to(continuous_action, relative=True, blocking=True)
-
-        # Sleep after sending the navigate command
-        rospy.sleep(0.5)
+            self.robot.nav.navigate_to(continuous_action, relative=True, blocking=True)
 
-        # After optionally moving - try to grasp
-        if try_to_grasp:
-            self.grasp_planner.go_to_manip_mode()
-            self.grasp_planner.try_grasping()
-
-    def set_goal(self, goal: str):
+    def set_goal(self, goal_find: str, goal_obj: str, goal_place: str):
         """Set the goal class as a string. Goal should be an object class we want to pick up."""
-        assert goal in self.goal_options
-        self.current_goal_id = self.goal_options.index(goal)
+        for goal in [goal_find, goal_obj, goal_place]:
+            assert goal in self.goal_options
+        goal_obj_id = self.goal_options.index(goal_obj)
+        goal_find_id = self.goal_options.index(goal_find)
+        goal_place_id = self.goal_options.index(goal_place)
+        self.task_info = {
+            "object_name": goal_obj,
+            "start_recep_name": goal_find,
+            "place_recep_name": goal_place,
+            "object_id": goal_obj_id,
+            "start_recep_id": goal_find_id,
+            "place_recep_id": goal_place_id,
+            "goal_name": f"{goal_obj} from {goal_find} to {goal_place}",
+            # Consistency - add ids for the first task
+            "object_goal": goal_obj_id,
+            "recep_goal": goal_find_id,
+        }
+        self.current_goal_id = self.goal_options.index(goal_obj)
         self.current_goal_name = goal
 
     def sample_goal(self):
         """set a random goal"""
-        # idx = np.random.randint(len(self.goal_options) - 2) + 1
-        idx = 2
-        self.current_goal_id = idx
-        self.current_goal_name = self.goal_options[idx]
+        goal_obj_idx = np.random.randint(len(self.goal_options))
+        goal_obj = self.goal_options[goal_obj_idx]
+        self.current_goal_id = self.goal_options.index(goal_obj_idx)
+        self.current_goal_name = goal_obj
 
     def get_observation(self) -> Observations:
         """Get Detic and rgb/xyz/theta from this"""
-        rgb, depth, xyz = self.get_images(compute_xyz=True, rotate_images=True)
+        rgb, depth, xyz = self.robot.head.get_images(
+            compute_xyz=True,
+        )
```

Comment:
Just double-checking that you've run this on the robot and PCDs looked fine with this argument change?

Reply:
yeah it looks fine